add fonts to related language addons

AlwinEsch opened the pull request:
### Description
This adds some TTF files in order to have the corresponding character type for upcoming changes in Kodi in the respective language.

The background is that Kodi does not support the characters in very many languages and could only be fixed in a few using "Arial.ttf".
Think about relocating it here, as it is needed in the respective language and so as not to flood Kodi when everyone was in it.

<!--- Provide a short summary of submitted add-on in case it's a new addition. -->
<!--- If it's plugin update only highlight biggest changes if needed. -->
<!--- Make sure you follow the checklist below before finalizing your pull-request. -->
### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-resources/blob/master/CONTRIBUTING.md) document
- [ ] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

AlwinEsch commented:
About the other fonts I created:
- https://github.com/AlwinEsch/resource.font.excluded
- https://github.com/AlwinEsch/resource.font.limited
- https://github.com/AlwinEsch/resource.font.active

Because of the "resource.font.active" has to be kept under observation a little (if moved to Kodi), if the associated language comes in here, it would have to be moved from there to here.

gade01 commented:
How will this work when skins provide the fonts?
Will these fonts provided with the language addons override the fonts provided with the skins?

AlwinEsch commented:
The skin is always as primary, only in case the wanted character becomes not found there, does it a fallback to this ones on language addons.

See on https://github.com/AlwinEsch/kodi/blob/fix-estuary-strings/xbmc/guilib/GUIFontTTF.cpp#L1231-L1233
The `m_fontMain` there relates to them where skin bring/selected.

gade01 commented:
@AlwinEsch That sounds really great.
I have been struggeling with adding fonts for several languages not using regular characters for years.
So this is a great solution! ;)

razzeee commented on noto_license.txt:
```diff
@@ -0,0 +1,48 @@
+Copyright (c) 2010-2014, Łukasz Dziedzic (dziedzic@typoland.com),
```
Do we really need this in the root folder too, you seem to include this same license in every addon folder next to the actual used font.

AlwinEsch replied:
Not really, will remove here as on addons are enough :sweat_smile:.

AlwinEsch commented:
Is updated and also added on second commit, additional font addons to allow Kodi in his use case the availability.

ProfYaffle commented on resource.font.limited/addon.xml:
```diff
@@ -0,0 +1,20 @@
+    point="kodi.resource.font"/>
+  <extension point="kodi.addon.metadata">
+    <summary lang="en_GB">Less common fonts</summary>
+    <description lang="en_GB">These written languages are still used in the world, but are used relatively little.</description>
```
Suggest something simpler, perhaps _Fonts for less widely spoken languages_?

ProfYaffle commented on resource.font.active/addon.xml:
```diff
@@ -0,0 +1,20 @@
+    point="kodi.resource.font"/>
+  <extension point="kodi.addon.metadata">
+    <summary lang="en_GB">Common fonts</summary>
+    <description lang="en_GB">More frequently used fonts, which Kodi cannot yet add to a language addon.</description>
```
Suggest something like _Common fonts that are not currently included in any of Kodi's language addons_

ProfYaffle commented on resource.font.excluded/addon.xml:
```diff
@@ -0,0 +1,20 @@
+    point="kodi.resource.font"/>
+  <extension point="kodi.addon.metadata">
+    <summary lang="en_GB">Mostly unused fonts in the world</summary>
+    <description lang="en_GB">This addon contains some font files, which bring formats that are largely unused, ancient and not widely used languages for example.</description>
```
Suggest _Fonts for languages that are widely considered to be largely unused, e.g. ancient scripts_ or something like that.

AlwinEsch committed:
add fonts to related language addons

This adds some TTF files in order to have the corresponding character type for upcoming changes in Kodi in the respective language.

The background is that Kodi does not support the characters in very many languages and could only be fixed in a few using "Arial.ttf".
Think about relocating it here, as it is needed in the respective language and so as not to flood Kodi when everyone was in it.

AlwinEsch commented on resource.language.zh_tw/addon.xml:
```diff
@@ -6,6 +6,7 @@
   provider-name="Team Kodi">
   <requires>
     <import addon="kodi.resource" version="1.0.0"/>
+    <import addon="resource.language.zh_cn" version="9.0.0"/> <!-- To get his font file about where support all -->
```
This is added new to get on `zh_tw` and `ja_jp` the font file from `zh_cn` where contains mostly all.

Bad is that on NotoSansJP-Regular.ttf and NotoSansTC-Regular.ttf also some where not on other, thats why also inside here another one.

gade01 commented:
@AlwinEsch What is required for this PR to get merged?
Core support?

AlwinEsch committed:
add font addons to allow other styles

With here are 3 font addons added:
- resource.font.active Fonts to active used characters
- resource.font.excluded Fonts to no more active charactes on world
- resource.font.limited Fonts to active, but with less usage
- resource.font.coloremoji Font about colored emoji's

AlwinEsch commented:
How we see the state about this? Maybe we can bring in and have ready for the request in Kodi?

@gade01 It can be come in before. The fonts inside here, then currently not used, but prepared for the request in Kodi (https://github.com/xbmc/xbmc/pull/20552)

gade01 commented:
It looks great to me, but perhaps I'm not the right person to review this. :)

enen92 commented:
@AlwinEsch please check addon-checker errors, it looks like the icons you specify on addon.xml do not exist.
I am also wondering if the font files should not be font addons instead, and those font addons specified as dependencies for the language addons

scott967 commented:
I tested these for the CJK variants back when the core PR was fresh and the language resources did work as expected.  Not sure anything gained by splitting out the fonts as the way the fonts are designed is language-specific.  Unless a skin/addon provides complete coverage in a skin-provided font, then the language font is not needed.    I think  https://github.com/xbmc/xbmc/pull/20552 needs a rebase to really test this again?

AlwinEsch commented:
> @AlwinEsch please check addon-checker errors, it looks like the icons you specify on addon.xml do not exist. I am also wondering if the font files should not be font addons instead, and those font addons specified as dependencies for the language addons

About the reason to combine the font with language is mainly to have on language switch immediately the needed font available, without it can come that the language is switched and user becomes asked for font and not readable then.
But I look with font add-on as dependency of language add-on, maybe it works.

I make on next days a rebase about the font pull requests and to become ready for use.

romanvm commented:
Is this still relevant?

romanvm commented:
@AlwinEsch is it still relevant or can be closed?

romanvm commented:
Last call before this PR will be closed as stale

romanvm commented:
Closing as stale
